test/jdk/sun/security/pkcs12/KeytoolOpensslInteropTest.java

```diff
@@ -52,28 +64,99 @@
 import static sun.security.util.KnownOIDs.*;
 import static sun.security.pkcs.ContentInfo.*;
 
-public class ParamsTest  {
+public class KeytoolOpensslInteropTest {
 
     public static void main(String[] args) throws Throwable {
-
-        // De-BASE64 textual files in ./params to `pwd`
-        try (DirectoryStream<Path> stream = Files.newDirectoryStream(
-                Path.of(System.getProperty("test.src"), "params"),
-                p -> !p.getFileName().toString().equals("README"))) {
-            stream.forEach(p -> {
-                try (InputStream is = Files.newInputStream(p);
-                     OutputStream os = Files.newOutputStream(p.getFileName())) {
-                    Base64.getMimeDecoder().wrap(is).transferTo(os);
-                } catch (IOException e) {
-                    throw new UncheckedIOException(e);
-                }
-            });
+        // openssl selection flow:
+        // 1. Check whether test.openssl.path is set and it's the the preferred
+        //    version(1.1.*) of openssl
+        // 2. If above property doesn't set, then look for already installed
+        //    openssl (version 1.1.*) in system path /usr/bin/openssl or
+        //    /usr/local/bin/openssl
+        // 3. if above is also not available try to download openssl from
+        //    artifactory
+        // If any of above 3 succeeds then perform all tests, otherwise skip
+        // all openssl command dependent tests.
+
+        String opensslPath = System.getProperty("test.openssl.path");
+        if (opensslPath == null) {
+            // checking the existence of already installed openssl
+            // in the test machine
+            opensslPath = getSystemOpensslPath();
         }
+        if (opensslPath == null) {
+            // trying to download from artifactory
+            opensslPath = OpensslArtifactFetcher.fetchOpenssl();
+        }
+        if (opensslPath != null && verifyOpensslVerion(opensslPath)) {
```

Comment:
The code here is a little different from your description above. If user provides a non-1.1.* openssl through `test.openssl.path` then you won't go look into elsewhere.

Also, `getSystemOpensslPath()` and `OpensslArtifactFetcher.fetchOpenssl()` already ensure version is 1.1.* so the check above is duplicated.

Reply:
Though I was kept it in intentional like that, to make sure to give proper openssl path if user is supplying it explicitly.
Anyhow I changed it now.
1. Check whether property test.openssl.path is set and it's the
    preferred version(1.1.*) of openssl, then return that path.
2. Else look for already installed openssl (version 1.1.*) in system
   path /usr/bin/openssl or /usr/local/bin/openssl, then return that
    path.
3. Else try to download openssl (version 1.1.*) from the artifactory
     and return that path, if download fails then return null.